crates/bevy_render/src/mesh/primitives/dim3/plane.rs

```diff
@@ -65,27 +78,66 @@ impl PlaneMeshBuilder {
         self.plane.half_size = Vec2::new(width, height) / 2.0;
         self
     }
+
+    /// Sets the subdivisions of the plane mesh.
+    ///
+    /// 0 - is the original plane geometry, the 4 points in the XZ plane.
+    ///
+    /// 1 - is split by 1 line in the middle of the plane on both the X axis and the Z axis,
+    ///     resulting in a plane with 4 quads / 8 triangles.
+    ///
+    /// 2 - is a plane split by 2 lines on both the X and Z axes, subdividing the plane into 3
+    ///     equal sections along each axis, resulting in a plane with 9 quads / 18 triangles.
+    #[inline]
+    pub fn subdivisions(mut self, subdivisions: u32) -> Self {
+        self.subdivisions = subdivisions;
+        self
+    }
 }
 
 impl MeshBuilder for PlaneMeshBuilder {
     fn build(&self) -> Mesh {
+        let z_vertex_count = self.subdivisions + 2;
+        let x_vertex_count = self.subdivisions + 2;
+        let num_vertices = (z_vertex_count * x_vertex_count) as usize;
+        let num_indices = ((z_vertex_count - 1) * (x_vertex_count - 1) * 6) as usize;
+
+        let mut positions: Vec<Vec3> = Vec::with_capacity(num_vertices);
+        let mut normals: Vec<[f32; 3]> = Vec::with_capacity(num_vertices);
+        let mut uvs: Vec<[f32; 2]> = Vec::with_capacity(num_vertices);
+        let mut indices: Vec<u32> = Vec::with_capacity(num_indices);
+
         let rotation = Quat::from_rotation_arc(Vec3::Y, *self.plane.normal);
-        let positions = vec![
-            rotation * Vec3::new(self.plane.half_size.x, 0.0, -self.plane.half_size.y),
-            rotation * Vec3::new(-self.plane.half_size.x, 0.0, -self.plane.half_size.y),
-            rotation * Vec3::new(-self.plane.half_size.x, 0.0, self.plane.half_size.y),
-            rotation * Vec3::new(self.plane.half_size.x, 0.0, self.plane.half_size.y),
-        ];
+        let size = self.plane.half_size * 2.0;
 
-        let normals = vec![self.plane.normal.to_array(); 4];
-        let uvs = vec![[1.0, 0.0], [0.0, 0.0], [0.0, 1.0], [1.0, 1.0]];
-        let indices = Indices::U32(vec![0, 1, 2, 0, 2, 3]);
+        for z in 0..z_vertex_count {
+            for x in 0..x_vertex_count {
+                let tx = x as f32 / (x_vertex_count - 1) as f32;
+                let tz = z as f32 / (z_vertex_count - 1) as f32;
+                let pos = rotation * Vec3::new((-0.5 + tx) * size.x, 0.0, (-0.5 + tz) * size.y);
+                positions.push(pos);
+                normals.push(self.plane.normal.to_array());
+                uvs.push([tx, tz]);
+            }
+        }
+
+        for y in 0..z_vertex_count - 1 {
```

Comment:
```suggestion
        for z in 0..z_vertex_count - 1 {
```
Maybe call this `z` as above?